librecad/src/actions/rs_actiondefault.cpp

```diff
@@ -109,16 +109,84 @@ void RS_ActionDefault::keyReleaseEvent(QKeyEvent* e) {
 }
 
 
+/*
+   Highlights hovered entities that are visible and not locked.
+
+   - by Melwyn Francis Carlo <carlo.melwyn@outlook.com>
+*/
+void RS_ActionDefault::highlightHoveredEntities(const RS_Vector& currentMousePosition)
+{
+    for (auto entity : *graphicView->getContainer())
+    {
+        if (entity != nullptr)
+        {
+            if (entity->isVisible() && ! entity->isLocked())
+            {
+                double hoverTolerance { 10.0 / graphicView->getFactor().magnitude() };
+
+                if (entity->rtti() == RS2::EntityEllipse)
+                {
+                    hoverTolerance /= 10.0;
+                }
+                else
+                {
+                    if (hoverTolerance < 1.0) hoverTolerance = 1.0;
+                }
+
+                bool isPointOnEntity = false;
+
+                if (((entity->rtti() >= RS2::EntityDimAligned) && (entity->rtti() <= RS2::EntityDimLeader)) 
```

Comment:
`RS2::EntityDim...`s and `RS2::EntityMText` don't have the `RS_Entity::isPointOnEntity` function on them. They use the `RS_EntityContainer::getNearestPointOnEntity` function instead.